### PR TITLE
Move AndMonoid to core and redefine Ior.Left and Ior.Right

### PR DESCRIPTION
**Pull request by i-walker**
Changes:
- Moving AndMonoid to core
- Redefine Ior.Left as Ior<A, Nothing> and Ior.Right as Ior<Nothing, A>. Inference resolution takes care of the rest.

### CODE REVIEW

File: modules/optics/arrow-optics/src/test/kotlin/arrow/optics/AndMonoidTest.kt [hunk 1]
```diff
@@ -1,14 +0,0 @@
-package arrow.optics
```
Comment (pakoito) [inline modules/optics/arrow-optics/src/test/kotlin/arrow/optics/AndMonoidTest.kt]: And this

Comment (pakoito) [reply modules/optics/arrow-optics/src/test/kotlin/arrow/optics/AndMonoidTest.kt]: Damn git I see it now

Comment (i-walker) [reply modules/optics/arrow-optics/src/test/kotlin/arrow/optics/AndMonoidTest.kt]: No Problem :)

File: modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt [hunk 2]
```diff
@@ -1018,7 +1018,7 @@ sealed class IO<out A> : IOOf<A> {
     override fun unsafeRunTimedTotal(limit: Duration): Option<A> = unsafeResync(this, limit)
   }
 
-  internal data class Bind<E, out A>(val cont: IO<E>, val g: (E) -> IO<A>) : IO<A>() {
+  data class Bind<E, out A>(val cont: IO<E>, val g: (E) -> IO<A>) : IO<A>() {
```
Comment (1Jajen1) [inline modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt]: Is this intended?

Comment (i-walker) [reply modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt]: Oh, that was laying around from the initial intent of the PR. Thank you @1Jajen1

Comment (i-walker) [inline modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt]:
```suggestion
  internal data class Bind<E, out A>(val cont: IO<E>, val g: (E) -> IO<A>) : IO<A>() {
```

Comment (pakoito) [reply modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt]: You can go to the file in the Files Changed tab, click on the `...` and select `Edit File`. AFAIK it should commit to this PR directly :D

**Pull request merged**